Split build component by service

Michkov opened the pull request:
Separate build component to:
- build-service
- jvm-build-service
- pipeline-service
- shared-resources
- build-templates

psturc commented on components/build/README.md:
```diff
@@ -1,113 +0,0 @@
-
-# App Studio/HACBS Build System
```
imo this readme file contains useful information - is none of it going to be needed in a future?

Michkov replied:
hmm, probably I could split it as well

Michkov replied:
just moved to build-service for now.

bnallapeta reviewed (approved):
LGTM. 
A minor comment:
Should be renamed from build-service to pipeline-service [here](https://github.com/Michkov/infra-deployments/blob/build-split/components/pipeline-service/tekton-results/.tekton/pipelines-as-code.yaml#L4-L5)?

Michkov commented:
> LGTM. A minor comment: Should be renamed from build-service to pipeline-service [here](https://github.com/Michkov/infra-deployments/blob/build-split/components/pipeline-service/tekton-results/.tekton/pipelines-as-code.yaml#L4-L5)?

Updated, moved under tekton-pipelines since pipeline-service namespace does not exist. But Anyway I am thinking to move these .tekton folders under one place and into one namespace in next PRs

adambkaplan commented:
LGTM from me.

tkdchen commented:
:+1:
